### PR TITLE
Implemented functionality to filter the orders on the basis of filter selected(#22qhq27)

### PR DESCRIPTION
**Pull request by ymaheshwari1**

### CODE REVIEW

File: src/store/modules/order/actions.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ const actions: ActionTree<OrderState, RootState> = {
       if (resp && resp.status === 200 && !hasError(resp)) {
         let orders = resp.data.grouped.orderId.groups.map((order: any) => {
           order.orderId = order.doclist.docs[0].orderId
-          order.customerName = order.doclist.docs[0].customerPartyName
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts:21]: We will follow
https://github.com/hotwax/commerce-hub/blob/main/src/types/Order.ts#L4
```
customer: {
  name:
  email:
  .....
} 
```

File: src/views/FindOrder.vue [hunk 2]
```diff
@@ -28,57 +28,82 @@
           <ion-list>
             <ion-list-header>{{ $t("Date") }}</ion-list-header>
             <ion-item>
-              <ion-label>order created</ion-label>
-              <ion-select value="any">
-                <ion-select-option value="any">any</ion-select-option>
-              </ion-select>
+              <ion-label>{{ $t("Order created") }}</ion-label>
+              <ion-chip id="open-order-created-date-modal" slot="end">
```
Comment (adityasharma7) [inline src/views/FindOrder.vue]: I think this is not action but an id
```suggestion
              <ion-chip id="order-created-date-modal" slot="end">
```

Comment (adityasharma7) [reply src/views/FindOrder.vue]: Same applies to other ids

File: src/views/FindOrder.vue [hunk 3]
```diff
@@ -314,10 +428,33 @@ export default defineComponent ({
...
+    },
+    orderCreationDateUpdated(ev: CustomEvent) {
+      this.appliedFilters.date.orderCreated = ev['detail'].value
+      this.getOrders();
+    },
+    orderAutoCancelDateUpdated(ev: CustomEvent) {
+      this.appliedFilters.date.autoCancelDate = ev['detail'].value
+      this.getOrders();
+    },
+    orderPromiseDateUpdated(ev: CustomEvent) {
```
Comment (adityasharma7) [inline src/views/FindOrder.vue:447]:
```suggestion
    updateOrderPromiseDate(ev: CustomEvent) {
```
Same applies to other methods

File: src/views/FindOrder.vue [hunk 4]
```diff
@@ -314,10 +428,33 @@ export default defineComponent ({
...
+      this.appliedFilters.date.promiseDate = ev['detail'].value
+      this.getOrders();
     }
   },
-  mounted() {
+  async mounted() {
     this.getOrders();
+    await this.store.dispatch('order/getPurchaseOrderIds').then(ids => {
```
Comment (adityasharma7) [inline src/views/FindOrder.vue]: If we are not performing any change to state, action may not be needed. We could directly call service and use the response. WDYT?

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/FindOrder.vue (hunk 2 above)
Comment (ymaheshwari1) [reply src/views/FindOrder.vue]: Sure Sir, will update the ids

File: src/views/FindOrder.vue (hunk 3 above)
Comment (ymaheshwari1) [reply src/views/FindOrder.vue:447]: Sir, changed this logic to use a single method `appliedFiltersUpdated` which in turn calls an action that prepares a query for us. #71

File: src/views/FindOrder.vue (hunk 4 above)
Comment (ymaheshwari1) [reply src/views/FindOrder.vue]: Yes Sir, removed the action and directly used the service to get the ids.

File: src/store/modules/order/actions.ts (hunk 1 above)
Comment (ymaheshwari1) [reply src/store/modules/order/actions.ts:21]: Sir, used the same pattern as defined in the order type.

**Pull request merged**